### PR TITLE
Add OpenTelemetry Collector Builder docs

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

## Docs PR Checklist

- [x] This PR is for a documentation page whose authoritative copy is in the
      opentelemetry.io repository.

---

**Preview**:

- https://deploy-preview-2516--opentelemetry.netlify.app/docs/collector/builder/
- https://deploy-preview-2516--opentelemetry.netlify.app/docs/collector/distributions/
- https://deploy-preview-2516--opentelemetry.netlify.app/docs/concepts/distributions/
- https://deploy-preview-2516--opentelemetry.netlify.app/docs/concepts/glossary/#builder

### CODE REVIEW

**Comment by jpkrohling** [comment]
I'm still unable to test the changes locally, so forgive me for wasting the CI cycles to test the changes. I would appreciate a review, but I'm yet to verify all the links work as expected. Other than that, this PR is ready.

**Comment by jpkrohling** [comment]
This is now ready for review. I would appreciate all your feedback, especially from people who are familiar with the builder (devs and/or users).

File: content/en/docs/collector/builder.md
Comment (mx-psi) [inline content/en/docs/collector/builder.md:1]: Should we mention how to get the builder in this document (i.e. link to the releases page)?

Comment (jpkrohling) [reply content/en/docs/collector/builder.md:1]: YES, what a thing to forget!

Comment (jpkrohling) [reply content/en/docs/collector/builder.md:1]: Added, PR updated.

**Comment by nicolastakashi** [comment]
Very nice and easy to read @jpkrohling!
IMHO the text is pretty clear and easy to understand so far.

**Review by frzifus — APPROVED** [review]
Awesome @jpkrohling !

**Review by mx-psi — APPROVED** [review]
:rocket: :building_construction:

File: content/en/docs/concepts/glossary.md
Comment (chalin) [inline content/en/docs/concepts/glossary.md:56]:
```suggestion
See [Collector Builder](#collector-builder).
```

File: content/en/docs/concepts/glossary.md
Comment (chalin) [inline content/en/docs/concepts/glossary.md:81]:
```suggestion
A command-line tool that generates sources and binaries for
OpenTelemetry Collector [distributions](/docs/collector/distributions/). For more information, see [OpenTelemetry Collector Builder (ocb)](/docs/collector/builder).
```

File: content/en/docs/concepts/glossary.md
Comment (chalin) [inline content/en/docs/concepts/glossary.md:207]:
```suggestion
See [Collector Builder](#collector-builder).
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:2]:
```suggestion
title: OpenTelemetry Collector Builder (OCB)
linkTitle: Builder
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:6]:
```suggestion
The OpenTelemetry Collector Builder (OCB) is a command-line tool that
```

I know that the tool itself is named `ocb`, but the acronym should be in uppercase. When referring to the tool, it is ok to use lowercase.

**Review by chalin — COMMENTED** [review]
Some initial comments inline. More might follow. Thanks!

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:13]:
```suggestion
users to cherry-pick components they need for their production usage,
potentially even including their own custom components.
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:22]:
```suggestion
Download `ocb` from the Collector's
[release page](https://github.com/open-telemetry/opentelemetry-collector/releases). 
A container image is **NOT** available at this time. We recommend that you download and use
the same version that you'll use in your manifest, since the builder is only
guaranteed to work with the OpenTelemetry Collector API matching the builder's
version.
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:59]:
```suggestion
```sh
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:57]:
```suggestion
Build a distribution by running the following command:
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:26]:
```suggestion
A central part of a custom distribution is the distribution's manifest. This
```

File: content/en/docs/collector/builder.md
Comment (codeboten) [inline content/en/docs/collector/builder.md:8]: Putting the purpose of the tool in the opening paragraph will help users understand when they should use the builder sooner.

```suggestion
The OpenTelemetry Collector Builder (ocb) is a command-line interface tool that
allows users to build a custom [distribution](../distributions) for the OpenTelemetry
Collector.
```

File: content/en/docs/collector/builder.md
Comment (codeboten) [inline content/en/docs/collector/builder.md:13]:
```suggestion
This tool is currently being used to build both the [core](https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol) and the [contrib](https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib)
distributions. It can be used to build custom distributions as well, allowing
users to cherry-pick which components they need for their production usage,
potentially even including their own custom components.
```

File: content/en/docs/collector/builder.md
Comment (codeboten) [inline content/en/docs/collector/builder.md:30]:
```suggestion
This is an example `manifest.yaml`, containing only four components:
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:89]:
```suggestion
If the previous configuration is saved to `sidecar.yaml`,
then you can start the custom distribution as follows:
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:91]:
```suggestion
```sh
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:98]:
```suggestion
one go. While this is recommended practice, there are situations where
```

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:100]:
```suggestion
generate the sources and further customize it. In such cases, use the
```

**Review by codeboten — COMMENTED** [review]
Nice! Just a few suggestions

File: content/en/docs/collector/distributions.md
Comment (chalin) [inline content/en/docs/collector/distributions.md:15]:
```suggestion
components. For the components included in a [distribution][distributions], see the distribution's `manifest.yaml` file.
```

File: content/en/docs/collector/distributions.md
Comment (chalin) [inline content/en/docs/collector/distributions.md:20]:
```suggestion
binary size of the contrib distribution, it might be intimidating to most users. Users of the
```

File: content/en/docs/collector/builder.md
Comment (seankhliao) [inline content/en/docs/collector/builder.md:37]: comments explaining how these values affect the output?

File: content/en/docs/collector/builder.md
Comment (seankhliao) [inline content/en/docs/collector/builder.md:24]: link to somewhere where all the keys for the manifest are defined / explained?

File: content/en/docs/collector/builder.md
Comment (seankhliao) [inline content/en/docs/collector/builder.md:55]: While slightly out of scope for building a collector, I'd like to see a minimal dockerfile for packaging the output a build.
This was the only other thing I looked for in the releases repo when I needed to build my own instance.
While simple enough to write on your own, it's nice to have a confirmation of the (lack of) required dependencies.

File: content/en/docs/collector/builder.md
Comment (seankhliao) [inline content/en/docs/collector/builder.md:15]: tooling pre-requisites, like `go`?

File: content/en/docs/collector/builder.md
Comment (seankhliao) [reply content/en/docs/collector/builder.md:55]: semi-related, cross platform builds?
also static / CGO_ENABLED=0 ?

File: content/en/docs/collector/builder.md
Comment (chalin) [inline content/en/docs/collector/builder.md:39]: Here and elsewhere in the document, replace the current version (0.73.0) by `{{% param collectorVersion %}}`:

```suggestion
  otelcol_version: {{% param collectorVersion %}}
```

File: content/en/docs/collector/builder.md
Comment (jpkrohling) [reply content/en/docs/collector/builder.md:91]: For my own learning, why is sh better than console here?

File: content/en/docs/collector/builder.md
Comment (jpkrohling) [reply content/en/docs/collector/builder.md:55]: Good idea, would you mind adding that as a follow-up improvement to this, once it's merged? Given the number of approvals on this already, I would prefer to have this improvement on its own PR.

File: content/en/docs/collector/distributions.md
Comment (jpkrohling) [reply content/en/docs/collector/distributions.md:13]: Great catch, thank you!

File: content/en/docs/collector/builder.md
Comment (jpkrohling) [reply content/en/docs/collector/builder.md]: Done!

File: content/en/docs/collector/builder.md
Comment (jpkrohling) [reply content/en/docs/collector/builder.md:28]: I think this doc would be the authoritative source of this information. Would you be open to opening a follow-up PR with that?

File: content/en/docs/collector/builder.md
Comment (jpkrohling) [reply content/en/docs/collector/builder.md]: Done!

File: content/en/docs/collector/builder.md
Comment (jpkrohling) [reply content/en/docs/collector/builder.md]: I had to revert this, as it was causing the prettier script to break the final snippet in places where it would render the final YAML as an invalid config. Is there a way to bypass the prettier for this snippet?

**Comment by jpkrohling** [comment]
This PR is now ready to be reviewed again.

File: content/en/docs/collector/builder.md
Comment (chalin) [reply content/en/docs/collector/builder.md:46]: Yes, place `<!-- prettier-ignore -->` before the start of the code block.

If you prefer, I can make this change after the PR is merged. Let me know.

File: static/refcache.json
Comment (chalin) [inline static/refcache.json:3029]: The OTel website shouldn't be in the refcache:

```suggestion
```

**Review by cartermp — APPROVED** [review]
LGTM modulo @chalin's feedback. There's a little bit of passive voice in here that could be changed, but it's minor and generally difficult to work around in docs like this. Thank you!

**Review by codeboten — APPROVED** [review]

**Comment by svrnm** [comment]
We have https://opentelemetry.io/docs/collector/custom-collector/ already, what's the difference/overlap?

**Comment by jpkrohling** [comment]
@svrnm, how could I have missed this! I think I was looking for a builder-specific document and didn't spot that page. Should I close this PR?

**Comment by svrnm** [comment]
> @svrnm, how could I have missed it! I think I was looking for a builder-specific document and didn't spot that page. Should I close this PR?

Let's keep it open for now, I would like to understand what the differences are and if we can improve the existing content using the one you provide here

**Comment by cartermp** [comment]
I've also missed it. Whew, thanks @svrnm.

@jpkrohling I know it's a big ask, but do you believe that the content you've authored is better? I'd rather replace it wholesale if that's the case, since existing content doesn't "win" just be being there already 🙂. Or there's perhaps there's chunks of each that's best to keep?

**Comment by jpkrohling** [comment]
I tried to compare both, and they are pretty similar. Some things asked here as present there already, like a list of flags and their description. But I would leave the decision on which is better to someone else, as I'm certainly influenced by the text I wrote :-)

**Comment by jpkrohling** [comment]
I'm closing this, as I don't see a significant advantage of this over the current page. I'll end up using this for the original purpose, though, which is a document outside of OTel's website, but feel free to get chunks (or the entirety) of this doc as you see fit.

**Pull request closed without merging**
